app/controllers/instructeurs/procedures_controller.rb

```diff
@@ -324,8 +325,48 @@ def apercu
       DossierPreloader.load_one(@dossier)
     end
 
+    def history
+      @procedure = procedure
+      all_published_revisions = @procedure.revisions.where.not(published_at: nil).order(published_at: :asc)
+
+      @comparison_pairs = []
+
+      if all_published_revisions.size >= 2
+        1.upto(all_published_revisions.size - 1) do |i|
+          previous = all_published_revisions[i - 1]
+          current = all_published_revisions[i]
+
+          @comparison_pairs << [current, previous]
+        end
+
+        @comparison_pairs.reverse!
```

Comment:
je pense que tu peux laisser a la vue le fait de pairer les revisions. cf prochain commentaire
```suggestion
      @all_published_revisions = @procedure.revisions.where.not(published_at: nil).order(published_at: :desc)
```

Reply:
je viens d'y penser. faut un `.reorder(published_at: :desc)` car le model `Procedure` ordonnance par défault par :id